flink-table/flink-table-planner/src/test/resources/org/apache/flink/table/planner/plan/nodes/exec/stream/MatchRecognizeJsonPlanTest_jsonplan/testMatch.out

```diff
@@ -427,7 +463,7 @@
         "cid" : "BIGINT"
       } ]
     },
-    "description" : "Match(orderBy=[proctime ASC], measures=[FINAL(A\".id) AS aid, FINAL(l.id) AS bid, FINAL(C.id) AS cid], rowsPerMatch=[ONE ROW PER MATCH], after=[SKIP TO NEXT ROW], pattern=[((_UTF-16LE'A\"', _UTF-16LE'l'), _UTF-16LE'C')], define=[{A\"==(LAST(*.$1, 0), _UTF-16LE'a'), l==(LAST(*.$1, 0), _UTF-16LE'b'), C==(LAST(*.$1, 0), _UTF-16LE'c')}])"
+    "description" : "Match(orderBy=[proctime ASC], measures=[FINAL(FINAL(A\".id)) AS aid, FINAL(FINAL(l.id)) AS bid, FINAL(FINAL(C.id)) AS cid], rowsPerMatch=[ONE ROW PER MATCH], after=[SKIP TO NEXT ROW], pattern=[((_UTF-16LE'A\"', _UTF-16LE'l'), _UTF-16LE'C')], define=[{A\"==(LAST(*.$1, 0), _UTF-16LE'a'), l==(LAST(*.$1, 0), _UTF-16LE'b'), C==(LAST(*.$1, 0), _UTF-16LE'c')}])"
```

Comment:
Why this plan change?

Reply:
It is initially because we add validation on the query of insert which is ignored before. 
I finally found out that the root cause and fixed in the latest commit: when converting sql node to operation, the query of insert would be validated twice and the validator (in SqlValidatorImpl#navigationInMeasure) rewrite the FINAL call twice.